### PR TITLE
Fix E2E tests in non-Vagrant environment

### PR DESCRIPTION
**Pull request by mbarnes**
This 1st commit gets the E2E tests partially working in a non-Vagrant environment.

All tests pass when using Vagrant, 10 tests fail when not.  I'm hoping to fix these without touching service code.
```
Failing scenarios:
  features/cluster/deploy.feature:28  Initiate tree deployment with authentication
  features/cluster/hosts.feature:79  Verifying a cluster after adding a host (1)
  features/cluster/hosts.feature:186  Deleting a host also removes it from its cluster (1)
  features/cluster/hosts.feature:199  Deleting a host also removes it from its cluster (2)
  features/cluster/hosts.feature:212  Deleting a host also removes it from its cluster (3)
  features/cluster/list.feature:26  List all clusters with authentication
  features/cluster/restart.feature:28  Initiate cluster restart with authentication
  features/cluster/upgrade.feature:28  Initiate cluster upgrade with authentication
  features/hosts/retrieve.feature:27  Retrieve existing host with authentication
  features/hosts/retrieve.feature:49  Retrieve existing host with authentication

6 features passed, 6 failed, 0 skipped
43 scenarios passed, 10 failed, 5 skipped
280 steps passed, 10 failed, 56 skipped, 0 undefined
Took 0m55.263s
```

### CODE REVIEW

File: features/environment.py [hunk 1]
```diff
@@ -342,22 +335,34 @@ def before_all(context):
         if context.config.userdata.get('start-storage-service'):
             context.PROCESSES['commissaire-storage-service'] = try_start(
                 start_commissaire_service, 'commissaire-storage-service',
-                context, ['commissaire-storage-service'])
+                context, [
```
Comment (ashcrow) [inline features/environment.py]: I think this service will need to copy the default and update the storage handler to point to the generated etcd port.

Comment (mbarnes) [reply features/environment.py]: Yeah you're right.  Think I have a solution.

**Comment by mbarnes** [comment]
@ashcrow :arrow_up:  All tests passing for me.

Now, any ideas how we might tie this into Travis?  The test environment requires multiple git repos, if only for the config files in `commissaire-http`.

**Comment by ashcrow** [comment]
I'll take a look tomorrow when I have time, but let's tie it in via a different PR.

**Comment by ashcrow** [comment]
testing now

File: features/environment.py [hunk 2]
```diff
@@ -340,24 +346,41 @@ def before_all(context):
...
 
         if context.config.userdata.get('start-storage-service'):
-            context.PROCESSES['commissaire-storage-service'] = try_start(
+            # Feed the service with a configuration through stdin.
+            process = try_start(
                 start_commissaire_service, 'commissaire-storage-service',
-                context, ['commissaire-storage-service'])
+                context, [
+                    'commissaire-storage-service',
+                    '--config-file', '-'],
```
Comment (ashcrow) [inline features/environment.py:354]: Ha! Good idea!

**Comment by ashcrow** [comment]
## Default

```
$ ./tools/behave
...
12 features passed, 0 failed, 0 skipped
50 scenarios passed, 0 failed, 8 skipped
290 steps passed, 0 failed, 56 skipped, 0 undefined
Took 0m24.318s
```

## With Vagrant
```
$ ./tools/vagrantup
...
$ ./tools/behave -D use-vagrant
...
12 features passed, 0 failed, 0 skipped
58 scenarios passed, 0 failed, 0 skipped
346 steps passed, 0 failed, 0 skipped, 0 undefined
Took 1m1.437s
```

**Review by ashcrow — APPROVED** [review]

**Pull request merged**